### PR TITLE
Adds the DPCT generated codes, Makefile (without clang++ -fsycl) and slurm submit script for CSD3

### PR DESCRIPTION
**Pull request by krishnakumarg1984**
- adds dpct generated sycl codes for cuda2 version
 - adds a Makefile (outdated). Needs to be updated with the nightly build of open-source dpcpp
 - adds a slurm submit Makefile
 - Updates to gitignore

### CODE REVIEW

File: tests/cuda2/dpct_output/Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+CC=dpcpp
+CFLAGS=-O2 -Wall
+LDFLAGS=
+
+INCLUDES=-Iincludes
+
+all: main
+
+main: main.c dw_cuda_soa.dp.cpp read_in.c
+	$(CC) $(INCLUDES) $(CFLAGS) $^ -o $@
```
Comment (tkoskela) [inline tests/cuda2/dpct_output/Makefile:10]: We will need to be able to use the Makefile to build with different compilers and architecture targets. I can use this one if I invoke it with

```
make CC=clang++ CFLAGS="-O2 -Wall -fsycl -fsycl-targets=nvptx64-nvidia-cuda"
```

but it feels a bit cumbersome. At the least we will need to document the commands we are using. Do you have any ideas how to improve it?

One way would be following [this stackoverflow thread](https://stackoverflow.com/questions/13358131/one-makefile-for-two-compilers) set `CC` and `CFLAGS` for each target and then have separate targets for `intel` and `nvidia`.

```suggestion
CC=dpcpp
CFLAGS=-O2 -Wall
LDFLAGS=
INCLUDES=-Iincludes

intel: CC = dpcpp

nvidia: CC = clang++
nvidia: CFLAGS += -fsycl -fsycl-targets=nvptx64-nvidia-cuda

all: intel nvidia

intel: main.c dw_cuda_soa.dp.cpp read_in.c
	$(CC) $(INCLUDES) $(CFLAGS) $^ -o main
	
nvidia: main.c dw_cuda_soa.dp.cpp read_in.c
	$(CC) $(INCLUDES) $(CFLAGS) $^ -o main	
```

**Review by tkoskela — CHANGES_REQUESTED** [review]
Thanks for opening the PR, I think we can close #13 once this is merged. It looks to me as if there are still a few extra files that could be cleaned up before merging.

I also added a suggestion to make the Makefile a bit more flexible, but I'm very open to other ideas as well.

We may have to think a bit about the directory structure. I think we can keep the source code and the Makefile in `dpct_output` for now, but I would keep job scripts and such in a separate directory. Perhaps create a directory called `extra` under `tests/cuda2`, what do you think?

File: run/input.in.txt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+nmx   1024
+res   1.0e-11
+
+[Wilson flow]
+integrator   RK3
+eps   1.0e-2
+nstep   400
+dnms   5
+
+
```
[the remark this answers is not in the document]
Comment (krishnakumarg1984) [reply run/input.in.txt:82]: Done.

File: extra/rename_bin_files.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+do
+    BASENAME=$(basename --suffix=".$EXTENSION" $fn)
+    mv $fn processed/$BASENAME-$1-$2-$3-$4
+done
+
+for fn in $(find . -maxdepth 1 -name "[a-z].$EXTENSION")
+do
+    BASENAME=$(basename --suffix=".$EXTENSION" $fn)
+    mv $fn processed/sp-$BASENAME-$1-$2-$3-$4
+done
```
[the remark this answers is not in the document]
Comment (krishnakumarg1984) [reply extra/rename_bin_files.sh]: Yes. Agree. Removed this file.

File: build/compile_settings.txt [hunk 4]
```diff
@@ -1,8 +1,8 @@
 CODELOC ..
 COMPILER mpicc
 MPI_INCLUDE ./
-CFLAGS -std=c99 -O2 -DAVX -DFMA3 -Werror -Wall
-LDFLAGS
+CFLAGS -std=c99 -Werror -Wall -g -Og
+LDFLAGS -g
```
[the remark this answers is not in the document]
Comment (krishnakumarg1984) [reply build/compile_settings.txt:5]: Yes. Maybe worth taking notice that LDFLAGS has a `-g` option passed to it. Reverted this change.

File: tests/cuda2/dpct_output/Makefile (hunk 1 above)
Comment (krishnakumarg1984) [reply tests/cuda2/dpct_output/Makefile:10]: I think it's best to write a CMakelists.txt that will handle this. Is it important to get to that before March 31, 2022?

**Comment by krishnakumarg1984** [comment]
> Perhaps create a directory called `extra` under `tests/cuda2`, what do you think?

Yes. I agree that the directory structure we currently have is not very intuitive (it's too deeply nested and not discoverable).

Putting stuff under `extra` maybe subtly conveys that it is not directly useful to the project? What kind of scripts should go into extras?

**Pull request merged**